src/Features/Core/Portable/Diagnostics/Analyzers/PreferFrameworkTypeDiagnosticAnalyzerBase.cs

```diff
@@ -0,0 +1,114 @@
+// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Collections.Immutable;
+using Microsoft.CodeAnalysis.CodeStyle;
+using Microsoft.CodeAnalysis.Options;
+
+namespace Microsoft.CodeAnalysis.Diagnostics.PreferFrameworkType
+{
+    internal abstract class PreferFrameworkTypeDiagnosticAnalyzerBase<TSyntaxKind, TExpressionSyntax, TPredefinedTypeSyntax> :
+        DiagnosticAnalyzer, IBuiltInAnalyzer
+        where TSyntaxKind : struct
+        where TExpressionSyntax : SyntaxNode
+        where TPredefinedTypeSyntax : TExpressionSyntax
+    {
+        private static readonly LocalizableString s_preferFrameworkTypeMessage =
+            new LocalizableResourceString(nameof(FeaturesResources.Use_framework_type),
+                                          FeaturesResources.ResourceManager, typeof(FeaturesResources));
+
+        private static readonly LocalizableString s_preferFrameworkTypeTitle =
+            new LocalizableResourceString(nameof(FeaturesResources.Use_framework_type),
+                                          FeaturesResources.ResourceManager, typeof(FeaturesResources));
+
+        private static readonly DiagnosticDescriptor s_descriptorPreferFrameworkTypeInDeclarations =
+            new DiagnosticDescriptor(
+                IDEDiagnosticIds.PreferFrameworkTypeInDeclarationsDiagnosticId,
+                s_preferFrameworkTypeTitle,
+                s_preferFrameworkTypeMessage,
+                DiagnosticCategory.Style,
+                DiagnosticSeverity.Hidden,
+                isEnabledByDefault: true);
+
+        private static readonly DiagnosticDescriptor s_descriptorPreferFrameworkTypeInMemberAccess =
+            new DiagnosticDescriptor(
+                IDEDiagnosticIds.PreferFrameworkTypeInMemberAccessDiagnosticId,
+                s_preferFrameworkTypeTitle,
+                s_preferFrameworkTypeMessage,
+                DiagnosticCategory.Style,
+                DiagnosticSeverity.Hidden,
+                isEnabledByDefault: true);
+
+        public bool RunInProcess => true;
```

Comment:
@balajikris what makes this need to run inproc currently.  I want to know just so we undrestand the work that we'll need to do to move this out of proc.  Is it just because you access options?